zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/store/ElasticsearchBatchRequestTest.java

```diff
@@ -434,10 +438,33 @@ void shouldThrowPersistenceExceptionIfAResponseItemHasError() throws IOException
 
     // When
     batchRequest.add(INDEX, entity);
-    final ThrowingCallable callable = () -> batchRequest.execute();
+    final ThrowingCallable callable = () -> batchRequest.execute(null);
 
     // Then
     assertThatThrownBy(callable).isInstanceOf(PersistenceException.class);
     verify(elasticsearchClient).bulk(any(BulkRequest.class));
   }
+
+  @Test
+  void shouldUseCustomErrorHandlerIfProvided() throws IOException {
+    // Given
+    final TestExporterEntity entity = new TestExporterEntity().setId(ID);
+
+    final BulkResponseItem item = mock(BulkResponseItem.class);
+    when(item.error()).thenReturn(new ErrorCause.Builder().reason("error").build());
+    when(item.index()).thenReturn(INDEX_WITH_HANDLER);
+
+    final BulkResponse bulkResponse = mock(BulkResponse.class);
+    when(bulkResponse.items()).thenReturn(List.of(item));
+
+    when(elasticsearchClient.bulk(any(BulkRequest.class))).thenReturn(bulkResponse);
+    final BiConsumer<String, Error> errorHandler = mock(BiConsumer.class);
+
+    // When
+    batchRequest.add(INDEX_WITH_HANDLER, entity);
+    batchRequest.execute(errorHandler);
+
+    // Then
+    verify(errorHandler).accept(eq(INDEX_WITH_HANDLER), any());
```

Comment:
🔧 We could also validate the error that is passed in, might be good to validate that we pass in the correct data.
```suggestion
    verify(errorHandler).accept(eq(INDEX_WITH_HANDLER), eq(new Error(...)));
```